src/Engine/Component/Trigger.cpp

```diff
@@ -0,0 +1,31 @@
+#include "Trigger.hpp"
+#include "../Trigger/SuperTrigger.hpp"
+
+using namespace Component;
+
+Trigger::Trigger() {
+    // ADD CONSTRUCTOR
+}
+
+Trigger::~Trigger() {
+
+    if(m_trigger != nullptr)
+        delete m_trigger;
+}
+
+Json::Value Trigger::Save() const {
+    Json::Value component;
+
+    component = m_trigger->Save();
+
+    return component;
+}
+
+SuperTrigger* Trigger::GetTrigger() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetTrigger' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)